### DISCUSSION TITLE
How do you test a renovate.json?

### DISCUSSION THREAD

**Discussion by alper** [renovatebot/renovate#29399] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hey,

It's incredibly unclear how this should work.

I can't really iterate on the `renovate.json` locally, or can I? It seems like I have to check it in and then wait for Mend to pick it up and run it. This is incredibly slow and frustrating.

If I look at last nights logs it seems like it hasn't picked up the config file in the repo root at all, which would explain some things but again it's entirely unclear how to trouble shoot this.

Am I doing something wrong or is it supposed to work this way?

**Comment by alper** [comment]
I'm trying to run it locally using something like:

```
docker run -e GITHUB_COM_TOKEN=ghp_REDACTED renovate/renovate
```

but that is immediately throwing:

```
FATAL: You must configure a GitHub token
 INFO: Renovate was run at log level "info". Set LOG_LEVEL=debug in environment variables to see extended debug logs.
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "t4YkRTkq8x2FJl3KWSlKa",
           "msg": "You must configure a GitHub token"
         }
       ]
```

Which does not seem very correct.

**Comment by alper** [comment]
In verbose it's saying it is finding the token, so I don't get the contradiction here:

```
DEBUG: Using RE2 regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in config.js
DEBUG: No config file found on disk - skipping
DEBUG: Converting GITHUB_COM_TOKEN into a global host rule
DEBUG: File config
       "config": {}
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ]
       }
DEBUG: Combined config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ]
       }
DEBUG: Enabling forkProcessing while in non-autodiscover mode
DEBUG: Found valid git version: 2.45.1
DEBUG: Setting global hostRules
DEBUG: Adding token authentication for github.com (hostType=github) to hostRules
FATAL: You must configure a GitHub token
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "J8FvU-JBTFffAisd_NRiB",
           "msg": "You must configure a GitHub token"
         }
       ]
```

**Reply by rarkins** [reply]
You need to set RENOVATE_TOKEN, not GITHUB_COM_TOKEN, just like is written at https://docs.renovatebot.com/modules/platform/github/. The latter is for people who are running on non-github platforms.

**Reply by alper** [reply]
I've gone through so much documentation and didn't find that.

**Comment by alper** [comment]
I don't get where it's getting these random config files and ignoring the one that's in the repo:

```
DEBUG: Using RE2 regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /runner/renovate/job_config.json
DEBUG: Detected config in env RENOVATE_CONFIG
```

Did I put that there?

**Reply by rarkins** [reply]
You selected at the top "Mend Renovate hosted app on github.com" but it seems that is wrong. Please describe exactly how you are running Renovate

**Reply by alper** [reply]
I have a Mend app installed in our org. Should I have selected something else?

Here's the job id: 86056cfe-a57c-432c-b0cb-6b66768d59f4

**Reply by rarkins** [reply]
Then where was this log from?

```
FATAL: You must configure a GitHub token
 INFO: Renovate was run at log level "info". Set LOG_LEVEL=debug in environment variables to see extended debug logs.
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "t4YkRTkq8x2FJl3KWSlKa",
           "msg": "You must configure a GitHub token"
         }
       ]
```

**Reply by alper** [reply]
That's me trying to run something locally to figure out what's happening.

**Comment by alper** [comment]
Oh wow, so I'm trying to iterate on this and closing all the PRs it generates, and now:

```
Because you closed this PR without merging, Renovate will ignore this update (3.22.3). You will get a PR once a newer version is released. To ignore this dependency forever, add it to the ignoreDeps array of your Renovate config.

If you accidentally closed this PR, or if you changed your mind: rename this PR to get a fresh replacement PR.
```

So if I get this working then I'll have to manually wipe this state somewhere?

**Reply by rarkins** [reply]
It's not a good idea to run Renovate two different ways on the same repo - such as the hosted app and self-hosted.

If you have PRs which block others like above, you will need to rename them (change the title), e.g. add a "-old" to the title, or simply "!".

**Reply by alper** [reply]
I'm totally lost here. Giving up with the Mend integration and seeing about running Renovate as a Github action.

Straight off the bat I see some conflicting documentation there as well but we'll see…

**Comment by alper** [comment, marked as answer]
The solution I had here was to merge the renovate config into `master`. That seems to be the ony reliable way to develop this.
